SDK.md

```diff
@@ -44,6 +44,68 @@ Supports CloudEvents spec milestones and ongoing development version.
 HTTP transport support (both structured and binary).
 Widely used programming language version.
 
+### Object model structure guidelines
+
+The CloudEvents specification set consists of
+
+* the base specification that defines the abstract information model and
+  associated rules for a CloudEvent,
+* a set of extensions to the base specification that define use-case
+  specific attributes and associated rules,
+* a set of event format specifications that define how a CloudEvent and  
+  its attributes are encoded in a particular wire transfer format, and
+* a set of transport bindings that define how a wire-encoded CloudEvent is
+  interoperably transported from a publisher or intermediary to a target
+  endpoint using a particular application protocol.
+
+The object model of any API projection of CloudEvents MUST reflect the
+specification architecture with the following elements:
+
+* A generic CloudEvent class/object/structure that provides access to the
+  CloudEvent attributes and enforces data type constraints and all further
+  conformance rules defined in the core CloudEvents specification. Instances
+  of this class/object/structure MUST always be in a conformant state.
+  The generic CloudEvent abstraction SHOULD provide strongly typed access
+  to all well-known attributes, and MUST also provide name-indexed access
```

Comment:
The C# SDK gives you name indexed access based on the set version of the event. The names change when you change the version. "subject" would thus be an extension in 0.1/0.2. This is exactly for evolving events over time, because with mandatory indexed access you can get at a "2.0" core attribute with the "1.0" SDK even if there's no strongly typed wrapper for it.